### PR TITLE
FpmSyncd and OrchAgent use redis pipeline

### PR DESCRIPTION
**Pull request by qiluo-msft**

### CODE REVIEW

File: fpmsyncd/fpmlink.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ using namespace swss;
 using namespace std;
 
 FpmLink::FpmLink(int port) :
-    m_bufSize(FPM_MAX_MSG_LEN * 2),
+    m_bufSize(FPM_MAX_MSG_LEN * 256),
```
Comment (stcheng) [inline fpmsyncd/fpmlink.cpp]: why 256? what does this number mean?

File: fpmsyncd/routesync.h [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #ifndef __ROUTESYNC__
 #define __ROUTESYNC__
 
+#include <memory>
```
Comment (stcheng) [inline fpmsyncd/routesync.h]: needed?

File: fpmsyncd/fpmlink.cpp (hunk 1 above)
Comment (stcheng) [inline fpmsyncd/fpmlink.cpp]: define this number somewhere? or why it is a good number?

File: orchagent/main.cpp [hunk 3]
```diff
@@ -198,6 +198,18 @@ int main(int argc, char **argv)
         exit(EXIT_FAILURE);
     }
 
+    SWSS_LOG_NOTICE("Enable redis pipeline");
+
+    attr.id = SAI_REDIS_SWITCH_ATTR_USE_PIPELINE;
```
Comment (lguohan) [inline orchagent/main.cpp]: >SAI_REDIS_SWITCH_ATTR_USE_PIPELINE; [](start = 14, length = 35)

where do we have this defined?

File: orchagent/orchdaemon.cpp [hunk 4]
```diff
@@ -77,6 +78,19 @@ bool OrchDaemon::init()
     return true;
 }
 
+/* Flush redis through sairedis interface */
+void OrchDaemon::flush()
+{
+    sai_attribute_t attr;
+    attr.id = SAI_REDIS_SWITCH_ATTR_FLUSH;
```
Comment (lguohan) [inline orchagent/orchdaemon.cpp]: >SAI_REDIS_SWITCH_ATTR_FLUSH [](start = 14, length = 27)

where do we have this defined?

File: orchagent/orchdaemon.cpp [hunk 5]
```diff
@@ -106,6 +120,7 @@ void OrchDaemon::start()
             for (Orch *o : m_orchList)
                 o->doTask();
 
+            flush();
```
Comment (lguohan) [inline orchagent/orchdaemon.cpp]: >flush(); [](start = 12, length = 8)

add comment here, something like below?

/* Let sairedis to flush all SAI function call to ASIC DB */

File: orchagent/orchdaemon.cpp [hunk 6]
```diff
@@ -77,6 +78,19 @@ bool OrchDaemon::init()
     return true;
 }
 
+/* Flush redis through sairedis interface */
+void OrchDaemon::flush()
+{
```
Comment (kcudnik) [inline orchagent/orchdaemon.cpp]: Add SWSS_LOG_ENTER()

**Pull request merged**